src/dbnode/storage/index/block.go

```diff
@@ -1257,10 +1259,16 @@ func (b *block) Tick(c context.Cancellable) (BlockTickResult, error) {
 		for _, seg := range group.segments {
 			result.NumSegments++
 			result.NumDocs += seg.Size()
+			// TODO(bodu): Revist this and implement a more sophisticated free strategy.
+			if immSeg, ok := seg.(segment.ImmutableSegment); ok {
+				if err := immSeg.FreeMmap(); err != nil {
```

Comment:
Note: since this change is kind of "mutative" perhaps we should take a `Lock()` and `Unlock()` in Tick now instead of an `RLock` like we currently do?